src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,262 @@
+        });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => {
+        if (column?.parse) {
+            row[column.name] = column.parse(event, row, column);
+        } else if ((event.target as any).type === "number") {
+            row[column.name] = Number(event.target.value);
+        } else {
+            row[column.name] = event.target.value;
+        }
+        if (rowCalculation) {
+            rowCalculation(data, row);
+        }
+        raiseEvent("custom", {
+            eventType: "cellChange",
+            rowData: { rows: data, row: row, column: column },
+        });
+        setData([...data]);
+        setValue(data);
+    };
+
+    return (
+        <Box maxHeight={maxHeight} maxWidth={maxWidth} sx={{ overflow: "auto" }}>
+            <Table
+                size={size}
+                stickyHeader={stickyHeader}
+                sx={{
+                    // Hack required to override .gcx-forms.defaults
+                    ["& span"]: {
+                        display: "inline-flex !important",
+                    },
+                }}
+            >
+                <TableHead>
+                    <TableRow>
+                        {includeFooter && <TableCell />}
+                        {cols.map((col) => (
+                            <TableCell align={col.align} key={col.name}>
+                                {col.alias ?? col.name}
+                            </TableCell>
+                        ))}
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {data.map((row, index) => {
+                        return (
+                            <TableRow hover key={index}>
+                                {includeFooter && <TableCell />}
+
+                                {cols.map((col, index) => {
+                                    const value = row[col.name];
+                                    const formattedValue = col.format ? col.format(value) : value;
+                                    return (
+                                        <TableCell
+                                            align={col.align}
+                                            key={index}
+                                            onMouseEnter={(event) =>
+                                                handleMouseEnter(event, row, col.name)
+                                            }
+                                            onMouseLeave={(event) =>
+                                                handleMouseLeave(event, row, col.name)
+                                            }
+                                        >
+                                            {col.editable ? (
+                                                <Input
+                                                    readOnly={false}
```

Comment:
The size and outlines on the inputs are a bit weird.

You could consider adding something like this to fix some of it `fullWidth={col.type === "text" ? true : false}` .

There are also a bunch of `.gcx-forms.defaults` styles on the inputs that are messing with the inputs. Those should be cleared.

I would get some advise from PXD on this.

Reply:
Ok I'll chat with them on Monday.